src/tstester/base.ts

```diff
@@ -2,10 +2,18 @@
 import { JDBus } from "../jdom/bus"
 import { assert } from "../jdom/utils"
 
+// A base class that abstracts away user-facing logging events, eg to pipe it to a web page
 export interface ConsoleUi {
     log: (msg: string) => void
 }
 
+// ConsoleUi that dumps to the debugging console
+export class DebugConsoleUi {
```

Comment:
you could also just pass a callback and ditch the class alltogether.

Reply:
did you mean to add ``implements ConsoleUi`` ?

Reply:
Fixed, oops